hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/nfttokeninfo/NftTokenInfoCallTest.java

```diff
@@ -0,0 +1,190 @@
+import static com.hedera.node.app.service.contract.impl.test.TestHelpers.EXPECTED_ROYALTY_CUSTOM_FEES;
+import static com.hedera.node.app.service.contract.impl.test.TestHelpers.FEE_SCHEDULE_KEY;
+import static com.hedera.node.app.service.contract.impl.test.TestHelpers.FREEZE_KEY;
+import static com.hedera.node.app.service.contract.impl.test.TestHelpers.FUNGIBLE_EVERYTHING_TOKEN;
+import static com.hedera.node.app.service.contract.impl.test.TestHelpers.KYC_KEY;
+import static com.hedera.node.app.service.contract.impl.test.TestHelpers.LEDGER_ID;
+import static com.hedera.node.app.service.contract.impl.test.TestHelpers.PAUSE_KEY;
+import static com.hedera.node.app.service.contract.impl.test.TestHelpers.SENDER_ID;
+import static com.hedera.node.app.service.contract.impl.test.TestHelpers.SUPPLY_KEY;
+import static com.hedera.node.app.service.contract.impl.test.TestHelpers.WIPE_KEY;
+import static com.hedera.node.app.service.contract.impl.test.TestHelpers.revertOutputFor;
+import static com.hedera.node.app.service.contract.impl.utils.ConversionUtils.headlongAddressOf;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+import com.esaulpaugh.headlong.abi.Tuple;
+import com.hedera.hapi.node.base.Key;
+import com.hedera.hapi.node.base.Timestamp;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.TokenTupleUtils.TokenKeyType;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.nfttokeninfo.NftTokenInfoCall;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.nfttokeninfo.NftTokenInfoTranslator;
+import com.hedera.node.app.service.contract.impl.test.exec.systemcontracts.hts.HtsCallTestBase;
+import com.hedera.node.config.data.LedgerConfig;
+import com.swirlds.config.api.Configuration;
+import java.math.BigInteger;
+import java.util.Collections;
+import java.util.List;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class NftTokenInfoCallTest extends HtsCallTestBase {
+    @Mock
+    private Configuration config;
+
+    @Mock
+    private LedgerConfig ledgerConfig;
+
+    @Test
+    void returnsNftTokenInfoStatusForPresentToken() {
+        when(config.getConfigData(LedgerConfig.class)).thenReturn(ledgerConfig);
+        when(ledgerConfig.id()).thenReturn(com.hedera.pbj.runtime.io.buffer.Bytes.fromHex(LEDGER_ID));
+        when(nativeOperations.getNft(FUNGIBLE_EVERYTHING_TOKEN.tokenId().tokenNum(), 2L))
+                .thenReturn(CIVILIAN_OWNED_NFT);
+
+        final var subject = new NftTokenInfoCall(mockEnhancement(), false, FUNGIBLE_EVERYTHING_TOKEN, 2L, config);
+
+        final var result = subject.execute().fullResult().result();
+
+        final var keyList = List.of(
```

Comment:
Maybe these two `keyList` constructors - for the real keys and the default keys - could be moved over to `TestHelpers` or somewhere?